src/frontend/src/components/LogIn/index.tsx

```diff
@@ -1,61 +1,192 @@
 import { logIn } from '../../redux/system/actions';
-import Logo from '../Logo';
 import { SystemState } from '../../redux/system/types';
+import apiService, { Request } from '../../service/api-service';
+import validateEmail from '../../service/email-validation';
 import history from '../History';
-import { Link } from 'react-router-dom';
+import Logo from '../Logo';
 
 interface Props {
   system: SystemState;
   logIn: typeof logIn;
 }
-interface State {}
+
+interface State {
+  email: string;
+  password: string;
+  invalidCredentials: boolean;
+  invalidResponse: boolean;
+  invalidEmail: boolean;
+  missingPassword: boolean;
+}
 
 class LogIn extends React.Component<Props, State> {
+  public state: State = {
+    email: '',
+    password: '',
+    invalidCredentials: false,
+    invalidResponse: false,
+    invalidEmail: false,
+    missingPassword: false
+  };
+
   constructor(props: Props) {
     super(props);
     this.logInNow = this.logInNow.bind(this);
   }
 
-  public logInNow(event: React.SyntheticEvent) {
-    this.props.logIn();
+  public handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+  };
+
+  public handleSubmit = (event: React.BaseSyntheticEvent) => {
     event.preventDefault();
     event.stopPropagation();
-    history.push('/oeci');
+
+    // Standardize email-lowerCase and trim both form inputs
+    this.setState(
+      {
+        email: this.state.email.toLowerCase().trim(),
+        password: this.state.password.trim()
+      },
+      () => {
+        this.validateForm();
+      }
+    );
+  };
+
+  public validateForm() {
+    // validate email returns true for email input of: "_@_._" empty returns false
+    this.setState(
+      {
+        invalidEmail: !validateEmail(this.state.email),
+        missingPassword: this.state.password.length === 0
+      },
+      () => {
+        // if no errors are present, logInNow()
+        if (!this.state.invalidEmail && !this.state.missingPassword) {
+          this.logInNow();
+        }
+      }
+    );
+  }
+
+  public logInNow() {
+    const request: Request = {
+      url: '/api/auth_token',
+      data: {
+        email: this.state.email,
+        password: this.state.password
+      },
+      method: 'post'
+    };
+
+    apiService(request)
+      .then(response => {
+        // attach token and userID to store
+        this.props.logIn(response.data);
+        history.push('/oeci');
+      })
+      .catch(error => {
+        if (error.response.status === 401) {
+          // error: email and password do not match
+          this.setState({
+            invalidCredentials: true
+          });
+        } else {
+          // error: technical difficulties
+          this.setState({
+            invalidResponse: true
+          });
+        }
```

Comment:
Optional change IMO but just as an FYI, you can write this if/else like this:

```
this.setState(
  error.response.status === 401 ?
    // error: email and password do not match
    { invalidCredentials: true } :
    // error: technical difficulties
    { invalidResponse: true }
)
```

Reply:
I do like the way this appears, in your previous change request you asked to make things more clear and I wasn't sure if leaving out the if/else contributed to that.

Also putting the `error.response.status...` in that place throws errors so I have gone for:

`        error.response.status === 401 ?  `
`        // error: email and password do not match  `
`        this.setState({invalidCredentials: true}) :  `
`        // error: technical difficulties  `
        `this.setState({invalidResponse: true});`
Which does look much more clean than the initial implementation!